Monika After Story/game/zz_dockingstation.rpy

```diff
@@ -2479,3 +2479,222 @@ label mas_dockstat_found_monika:
+            jump expression mas_farewells.dockstat_rtg_label
+        #Otherwise, we don't have a valid label and need to jump to a generic ready to go
+        jump mas_dockstat_generic_rtg
+
+
+    #Pop hist so we don't have more than one 'give me a second to get ready' in hist.
+    $ _history_list.pop()
+
+    # display menu options
+    # 4 seconds seems decent enough for waiting.
+    show screen mas_background_timed_jump(4, "mas_dockstat_generic_iowait")
+    menu:
+        "Hold on a second!":
+            hide screen mas_background_timed_jump
+            $ persistent._mas_dockstat_cm_wait_count += 1
+
+    # fall thru to the wait wait flow
+    if renpy.has_label(mas_farewells.dockstat_wait_menu_label):
+        call expression mas_farewells.dockstat_wait_menu_label
+    else:
+        #Otherwise fallback to this label
+        call mas_dockstat_generic_wait_label
+
+    #If we returned True, that means we cancelled
+    if _return:
+        #We need to clear all the vars in case we go dockstat again
+        $ mas_farewells.resetDockstatFlowVars()
+
+        #And now, we return what the generic label would have returned, that is if it's a string.
+        if isinstance(_return, basestring):
+            return _return
+        #This means we got a boolean and we can't return it because the event handler requires strings
+        return
+
+    # by default, continue looping
+    jump mas_dockstat_generic_iowait
+
+
+#GENERIC WAIT LABEL
+label mas_dockstat_generic_wait_label:
+    menu:
+        m "What is it?"
+        "Actually, I can't take you right now.":
+            call mas_dockstat_abort_gen
+
+            #Show Monika again
+            call mas_transition_from_emptydesk("monika 1ekc")
+            call mas_dockstat_abort_post_show
+
+            if renpy.has_label(mas_farewells.dockstat_cancel_dlg_label):
+                jump expression mas_farewells.dockstat_cancel_dlg_label
+
+            #Fallback to generic cancel
+            jump mas_dockstat_generic_cancel
+
+        "Nothing.":
+            # if we get here, we should jump back to the top so we can
+            # continue waiting
+            m 2hub "Oh, good! Let me finish getting ready."
+            return
+
+
+#GENERIC RTG LABEL
+#FOR CUSTOM FLOWS, REQUIRES THE FOLLOWING GLOBAL VARIABLE:
+# - mas_farewells.dockstat_failed_io_still_going_ask_label
+#If not set, the generic label will be used
+label mas_dockstat_generic_rtg:
+    # io thread should be done by now
+    $ moni_chksum = promise.get()
+    $ promise = None # clear promise so we dont have any issues elsewhere
+    call mas_dockstat_ready_to_go(moni_chksum)
+    if _return:
+        python:
+            persistent._mas_greeting_type = mas_idle_mailbox.get_ds_gre_type(
+                store.mas_greetings.TYPE_GENERIC_RET
+            )
+
+        call mas_transition_from_emptydesk("monika 1eua")
+
+        #Otherwise we just use the normal outro
+        m 1eua "I'm ready to go."
+        return "quit"
+
```

Comment:
Should reset the vars here too?
```suggestion

        #We need to clear all the vars in case we go dockstat again
        $ mas_farewells.resetDockstatFlowVars()

```